peers/topicpool.go

```diff
@@ -101,12 +109,32 @@ func (t *TopicPool) addToPendingPeers(peer *peerInfo) {
 		peerInfo: peer,
 		index:    notQueuedIndex,
 	}
+
+	if t.maxPendingPeers == 0 || t.maxPendingPeers > len(t.pendingPeers) {
```

Comment:
Plus unit test for this one maybe.

Reply:
`0` means unbound correct? If so we should probably add a comment explaining how values are interpreted

Reply:
This is kind of a point of contention that I should have left a PR comment on.

For some reason the tests expicitly set limits to 0, both upper and lower bounds, so I'm not sure what the right thing to do is here.

On one hand, I want unbounded to be an option still, on the other hand, 0 is a magic value that I'm not fond of and maybe we should update the tests.

What do you two think?

Reply:
I feel in general `-1` is better to specify no limits, as there's no ambiguity, while 0 might mean literally "don't have any", although in this particular case is a bit painful to have exactly `-1` as it's multiplied (could be less than 0 though), but I don't have a strong opinion, both 0 or -1 is fine by me, as long as there's a comment/test

Reply:
Do we update tests too to specify limits above 0? Some of these tests actually rely on nodes being remembered _while_ setting limits at 0.